### PR TITLE
Automate e2e tests and tags for fips build

### PR DESCRIPTION
**Pull request by hellozee**
- Adds REPO_ROOT for 1.8 tests
 - Automate tests pipeline for fips builds
 - Automate creating fips tags during fork sync
 - Disable failing tests with patches

### CODE REVIEW

File: tetrateci/fetch_all_tags.sh [hunk 1]
```diff
@@ -29,12 +29,23 @@ for tag in $tags; do
     if [[ ! $(git rev-parse --verify --quiet origin/tetrate-release-$branch) ]]; then
         # create the tetrate release branch if it doesn't exist with the workflows
         git checkout -b tetrate-release-$branch origin/tetrate-workflow
+        git merge $tag --no-edit --allow-unrelated-histories
+        git tag test-$tag-tetrate-v0
+
+        git checkout -b tetratefips-release-$branch origin/tetrate-workflow
```
Comment (deva26) [inline tetrateci/fetch_all_tags.sh:35]: Can we put fips in a separate if else block? Something similar to line 29 and 38 for non fips.

File: .github/workflows/e2e_tests.yml [hunk 2]
```diff
@@ -73,9 +81,14 @@ jobs:
 
       - name: set up go 1.15
         uses: actions/setup-go@v1
+        if: ${{ !contains(github.event.ref, 'fips') }}
         with:
           go-version: 1.15
 
+      - name: setup fips compliant go
```
Comment (deva26) [inline .github/workflows/e2e_tests.yml]: Is this needed? Do the tests fail without this? If so, we need to know the list of tests failing as it is likely to translate into istioctl not being compatible with fips istio.

File: tetrateci/test_1.7.sh [hunk 3]
```diff
@@ -36,7 +37,7 @@ go test -count=1 -timeout 30m ./tests/integration/mixer/telemetry/... -p 1 -test
 
 go test -count=1 ./tests/integration/telemetry/outboundtrafficpolicy -p 1 -test.v
 go test -count=1 -timeout 30m ./tests/integration/telemetry/. -p 1 -test.v
-go test -count=1 -timeout 30m ./tests/integration/telemetry/stats/... -p 1 -test.v
+go test -count=1 -timeout 30m ./tests/integration/telemetry/stats/... -p 1 -test.v -run "TestIstioCtlMetrics|TestStatsFilter|TestSetup|TestWasmStatsFilter|TestTcpMetric"
```
Comment (deva26) [inline tetrateci/test_1.7.sh]: What is the exception here

File: .github/workflows/e2e_tests.yml (hunk 2 above)
Comment (hellozee) [reply .github/workflows/e2e_tests.yml]: This got removed, I think you are checking an old revision

File: tetrateci/test_1.7.sh (hunk 3 above)
Comment (hellozee) [reply tetrateci/test_1.7.sh]: This got removed too, 😅

File: tetrateci/fetch_all_tags.sh (hunk 1 above)
Comment (hellozee) [reply tetrateci/fetch_all_tags.sh:35]: Ok

**Review by deva26 — APPROVED** [review]
Thanks, Kuntal

**Pull request merged**